lib/sms/sms.c

```diff
@@ -61,77 +70,97 @@ struct sms_subscriber {
 /** @brief List of subscribers. */
 static struct sms_subscriber subscribers[CONFIG_SMS_MAX_SUBSCRIBERS_CNT];
 
-/** @brief Check if the received AT notification is a +CMT event. */
-static bool sms_is_cmd_notification(const char *const at_notif)
-{
-	if (at_notif == NULL) {
-		return false;
-	}
-
-	if (strlen(at_notif) <= AT_SMS_NOTIFICATION_LEN) {
-		return false;
-	}
-
-	return (strncmp(at_notif, AT_SMS_NOTIFICATION,
-			AT_SMS_NOTIFICATION_LEN) == 0);
-}
-
-/** @brief Parse the +CMT unsolicited received message in PDU mode. */
-static int sms_cmt_notif_parse(const char *const buf)
+/** @brief Save the SMS notification parameters. */
+static int sms_cmt_at_parse(const char *const buf, struct sms_data *cmt_rsp)
 {
-	/* Parse the received message. */
 	int err = at_parser_max_params_from_str(buf, NULL, &resp_list,
 						AT_CMT_PARAMS_COUNT);
-
 	if (err != 0) {
 		LOG_ERR("Unable to parse CMT notification, err=%d", err);
 		return err;
 	}
 
-	/* Check the AT response format. */
-	if (at_params_valid_count_get(&resp_list) != AT_CMT_PARAMS_COUNT) {
-		LOG_ERR("Invalid CMT notification format");
-		return -EAGAIN;
+	if (cmt_rsp->alpha != NULL) {
+		k_free(cmt_rsp->alpha);
```

Comment:
Do you expect that there is allocated memory here?

Reply:
Yep, there is memory allocated except for the very first received SMS. This is not something that has really changed compared to NCS version of smslib. However, I'll still do some clean up work for the memory handling in this file